### DISCUSSION TITLE
Remove NNTP support

### DISCUSSION THREAD

**Discussion by gahr** [neomutt/neomutt#3966] [Development]
Our NNTP backend is neglected. To the best of my knowledge, none of the people actively working on NeoMutt uses it daily.  Bugs are piling up (#3858, #3791, #3766) and I don't think anybody is interested on working on that part of the code.

My proposal is to remove support for NNTP from NeoMutt. :)

**Comment by vuori** [comment]
This has been discussed on irc a couple of times, and in broad terms I agree. To reiterate my prior opinion:

On one hand it's sad because it works _mostly_ ok and seems to have _some_ users. But on the other there's quite a bit of non-obvious impedance mismatch between mail and news that makes the NNTP backend a hexagonal peg in a round hole wrt the rest of the codebase (witness all the `#ifdef USE_NNTP` spread through the code outside the backend itself).

Maybe we should post a call to the mailing lists/Reddit/README for a new NNTP maintainer together with a deprecation notice, and proceed with removal if no one comes forward? (This is of course made more difficult by the fact that probably not just anyone can take on the NNTP code and get productive in a reasonable timeframe: the problems aren't really in `nntp/*.c`, but the integration, so they would have to learn the _whole_ codebase and think about how to untangle the current morass of special cases.)

**Reply by vuori** [reply]
As a warning shot, we could make NNTP support a configure option that's disabled by default. That would hopefully bring forth people who care about it. (Related config options should be a no-op so that if you set up NNTP in 2015 and haven't used it since, your neomutt will still start up.)
